modules/MiniFigDirectionDetector.cpp

```diff
@@ -0,0 +1,152 @@
+/**
+ * @file   MinifigDirectionDetector.cpp
+ * @brief  ミニフィグの向きを検出するクラス
+ * @author nishijima515
+ */
+
+#include "MiniFigDirectionDetector.h"
+
+using namespace cv;
+using namespace dnn;
+
+MiniFigDirectionDetector::MiniFigDirectionDetector(const std::string& modelPath)
+{
+  // コンストラクタの実装
+  // ネットワークの読み込み
+  net = readNetFromONNX(modelPath);  // モデルのパスを設定する
+  if(net.empty()) {
+    std::cerr << "モデルの読み込みに失敗しました！" << std::endl;
+  } else {
+    std::cout << "モデルが正常に読み込まれました。" << std::endl;
+  }
+  net.setPreferableBackend(DNN_BACKEND_OPENCV);  // OpenCVバックエンドを使用
+  net.setPreferableTarget(DNN_TARGET_CPU);       // CPUをターゲットに設定
+}
+
+MiniFigDirectionResult MiniFigDirectionDetector::detect(const cv::Mat& frame,
+                                                        const std::string& saveImagePath)
+{
+  MiniFigDirectionResult result;
```

Comment:
resultは複数動作クラスで共有すると思うから、引数で受け取るようにした方がいいかな。あと、画像処理結果画像を格納するパスは固定でよい。もし不都合があれば理由を教えてくれると助かります。
```suggestion
void MiniFigDirectionDetector::detect(const cv::Mat& frame,
                                                        MiniFigDetectionResult& result)
{
```